### PR TITLE
Introduce batching callback to avoid sync flush in React

### PR DESCRIPTION
**Pull request by dvoytenko**
Per https://blog.logrocket.com/simplifying-state-management-in-react-apps-with-batched-updates/

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -76,7 +76,7 @@
     "require-await": "error",
     "rest-spread-spacing": ["error", "never"],
     "react/prop-types": "error",
-    "react-hooks/exhaustive-deps": ["error", {"additionalHooks": "useSelect"}],
+    "react-hooks/exhaustive-deps": ["error", {"additionalHooks": "useSelect|useBatchingCallback"}],
```
Comment (swissspidy) [inline .eslintrc]: While you're at it, could you actually remove the `useSelect` bit? It's a leftover from the Gutenberg era.

Comment (dvoytenko) [reply .eslintrc]: done

Comment (dvoytenko) [reply .eslintrc]: done

Comment (swissspidy) [reply .eslintrc]: Thanks!

**Review by swissspidy — APPROVED** [review]

**Pull request merged**